### PR TITLE
HalfSizeRSS

### PR DESCRIPTION
**Pull request by linuxgurugamer**
created by new automated script
hand verified
Closes #4284

### CODE REVIEW

File: NetKAN/HalfSizeRSS.netkan [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "license": "CC-BY-NC-SA",
+  "identifier": "HalfSizeRSS",
+  "x_via": "Automated SpaceDock CKAN submission",
+  "$kref": "#/ckan/spacedock/",
```
Comment (politas) [inline NetKAN/HalfSizeRSS.netkan]: This is unlikely to work

**Comment by linuxgurugamer** [comment]
Bug in script, fixed, regenerated and updated

**Comment by linuxgurugamer** [comment]
#rebuild

**Comment by politas** [comment]
This going to need at least one more netkan created for the Half RSS textures, so that can be a dependency.

**Comment by linuxgurugamer** [comment]
Look like he has 4 files in that zip, including the basic HalfRSS netkan.  Should I put them in all one pull?

**Comment by linuxgurugamer** [comment]
I've taken his files and fixed the syntax errors, also merged his basic .netkan with mine (he was m issing several dependencies).  So now I'm just waiting to hear the best way to do this.

**Comment by ayan4m1** [comment]
It looks like this mod has other dependencies hard-included in its SpaceDock download. I'm still learning about NetKAN packaging but I think Kopernicus and ModuleManager should be dependencies rather than copy-pasted into the mod ZIP.

Also, just checking where the mod is supposed to be sourced from - GitHub or SpaceDock - because right now the author-supplied netkans and the ones you have provided disagree over the source location for the mod. Both locations are including other mods without specifying them as dependencies.

**Comment by linuxgurugamer** [comment]
The version I have on my system has all the dependencies, and for now, I'm using the author-supplied source location.
The reason Kopernicus & ModuleManager are in the zip is for those people who aren't using  CKAN.  CKAN ignores those directories and my netkan properly includes them as dependencies.
Again, I'm waiting on advice on whether I should put all 4 netkan files into a single commit or not.

**Comment by politas** [comment]
So, I gather their are different netkans for the different texture quality levels - We would want them all to have a "provides" line and a "conflicts" section for a common "HalfRSSTextures" virtual netkan, and then have the main HalfRSS.netkan depend on that virtual netkan.

**Comment by linuxgurugamer** [comment]
I KNOW THAT!

My question is how to submit, as a single commit or multiple commits

**Comment by politas** [comment]
Oh, sorry. Don't think it really matters, as long as they're added to this PR's branch.

**Comment by linuxgurugamer** [comment]
ok.  I'll add these as one commit later this evening

**Comment by DinCahill** [comment]
CI failure incoming!

**Comment by DinCahill** [comment]
I don't actually know how to fix the error, though. I just ran NetKAN quicker than Jenkins. You might want to do the same to save time. You can also check the JSON with jsonlint.

**Comment by linuxgurugamer** [comment]
Not sure what the problem is with the HalfRSS.netkan, can someone take a look?

**Comment by techman83** [comment]
@DinCahill yeah, Jenkins has to spin up a container and it does a little more than just run NetKAN on the file :smile: 

@linuxgurugamer it's barfing on this NetKAN
```
Running NetKAN for NetKAN/HalfRSS-Textures2048.netkan
...
4738 [1] FATAL CKAN.NetKAN.Program (null) - After parsing a value an unexpected character was encountered: ". Path 'DOWNLOAD', line 5, position 3.
```

**Comment by DinCahill** [comment]
Should we run NetKAN through the debugger to see what's happening? I haven't done that with C# before, so I'll just do something else for now :P

**Comment by linuxgurugamer** [comment]
@techman83 It isn't barfing on that.  I have been running netkan.exe locally, it's barfing on the HalfRSS.netkan:
$ netkan.exe HalfRSS.netkan
544 [1] FATAL CKAN.NetKAN.Program (null) - After parsing a value an unexpected character was encountered: ". Path 'DOWNLOAD', line 5, position 3.

Found the problem, the .version file is in the wrong place.  Once I removed the vref line, it works

**Comment by linuxgurugamer** [comment]
Ok.  I've opened an issue with @pap1723 about the bad .version file, hopefully when he fixes it he will let me know and I'll update the netkan.
All 4 netkans are passing my local netkan.exe check, i believe the problem now is that the each depend on each other, and it's a race condition.  I suspect it will need manual intervention at this point

**Comment by techman83** [comment]
@linuxgurugamer aha - cool. The CI process puts all the metadata in the same PR into the local repository it creates. So it should be available. The build failure analyser reports this:

```
At least one GameData directory has been found within the GameData or a directory within a mods Installed Directory.
```
It also reports 
```
HalfRSS requires HalfRSS-Textures but nothing provides it
```
Which if you've dropped the `$vref` may be version related.

**Comment by politas** [comment]
No, it's just that netkan's testing each .netkan file separately rather than all at once, so the interlinked dependencies are doomed to failure. Let me test the install locally.

**Comment by linuxgurugamer** [comment]
The alternetive would have been to them up without any dependencies, and then update once accepted.
Thanks for testing locally

**Comment by politas** [comment]
Yep. Missing comma on the "DOWNLOAD" line in HalfRSS.version
That'd do it!

**Comment by techman83** [comment]
@politas not quite. It [generates](https://github.com/KSP-CKAN/xKAN-meta_testing/blob/master/NetKAN/build.sh#L306), then [injects](https://github.com/KSP-CKAN/xKAN-meta_testing/blob/master/NetKAN/build.sh#L345) all the build files before running [ckan](https://github.com/KSP-CKAN/xKAN-meta_testing/blob/master/NetKAN/build.sh#L371)

**Comment by linuxgurugamer** [comment]
@politas That's why I removed the vref

**Comment by politas** [comment]
Yep, just confirmingthat is what was wrong. All files test fine locally. ckan's created, all install fine.
Merging with thanks!

**Comment by politas** [comment]
@techman83, Ah, could it be because there are multiple providers of the dependency, and it needs input to choose one?

**Comment by linuxgurugamer** [comment]
Great.  I was chatting with the author earlier today, once he gets the version file fixed, I'll update the netkan for HalfRSS

**Comment by pap1723** [comment]
Here is the corrected .VERSION file for you. Thanks for all of the help getting this together.
[HalfSizeRSS.zip](https://github.com/KSP-CKAN/NetKAN/files/400218/HalfSizeRSS.zip)

**Comment by linuxgurugamer** [comment]
@pap1723 You need to get the corrected .version file into the release on giithub (and Spacedoc), Netkan/CKAN download from there.  The current config is downloading from the HalfSizeRSS release directory on Github

**Comment by techman83** [comment]
@politas aha, yes that'll do it!

**Comment by techman83** [comment]
@linuxgurugamer @pap1723 once that's done ping me and I'll nudge the indexer.

**Comment by pap1723** [comment]
OK, files are updated at Spacedock and Github!

**Comment by linuxgurugamer** [comment]
@techman83 I just updated the netkan

**Comment by linuxgurugamer** [comment]
or not :D

**Comment by linuxgurugamer** [comment]
I updated my branch, then realized you had merged.

**Comment by linuxgurugamer** [comment]
@techman83 Can you add the vref to the netkan and redo it?  If you need me to update it, let me know

**Comment by linuxgurugamer** [comment]
NM, I just did it on a new branch/pull

**Pull request merged**